indra/llcommon/llcoros.cpp

```diff
@@ -61,6 +61,28 @@
 #include <excpt.h>
 #endif
 
+// static 
+bool LLCoros::on_main_coro()
+{
+    if (!LLCoros::instanceExists() || LLCoros::getName().empty())
+    {
+        return true;
+    }
+    
+    // error out all the time
+    // This should only be called from inside llassert or similar
+    // and is low level enough that we can't rely on LLError
+    std::cerr << "LLCoros::on_main_coro(): not on a coroutine\n";
```

Comment:
Removed the assumption that it will only be called from llassert, but not doing assert_on_main_coro.  I suspect llassert(on_main_coro()) will show up in thousands of spots over time, and I want all of those compiled to no-ops.